domain/access/state/permission.go

```diff
@@ -722,13 +722,13 @@ func (st *PermissionState) upsertPermissionAuthorized(
 	// Does the apiUser have superuser access?
 	// Is permission the apiUser has on the target Admin?
 	authQuery := `
-WITH    ctrl AS (SELECT 'controller' AS c)
+WITH    ctrl AS (SELECT uuid FROM controller)
 SELECT  (p.grant_to, p.access_type) AS (&dbPermission.*)
 FROM    v_permission p
         JOIN v_user_auth u ON u.uuid = p.grant_to
-        LEFT JOIN ctrl ON p.grant_on = ctrl.c
+        LEFT JOIN ctrl ON p.grant_on = ctrl.uuid
 WHERE   u.name = $permInOut.name
-AND     (p.grant_on = $permInOut.grant_on OR p.grant_on = ctrl.c)
+AND     (p.grant_on = $permInOut.grant_on OR p.grant_on = ctrl.uuid)
 AND     u.disabled = false
 AND     u.removed = false
```

Comment:
I think the SQL can be reduced to:
```sql
SELECT  (p.grant_to, p.access_type) AS (&dbPermission.*)
FROM    v_permission p
        JOIN v_user_auth u ON u.uuid = p.grant_to
WHERE   u.name = $permInOut.name
AND     p.grant_on = $permInOut.grant_on
AND     u.disabled = false
AND     u.removed = false
```

The supplied grantOn string will always be a uuid now. We're checking permissions on the UUID provided, be it a model, controller or what not.

Reply:
Sweet, that makes the SQL a lot nicer

Reply:
Actually, just remembered from the test failures from doing this. This is on purpose. Here we are checking if the apiUser has access on the thing OR the controller, to see if they have permission to upsert. 

I've changed it to this which is a bit nicer:
```sql
SELECT  (p.grant_to, p.access_type) AS (&dbPermission.*)
FROM    v_permission p
        JOIN v_user_auth u ON u.uuid = p.grant_to
        LEFT JOIN controller c ON p.grant_on = c.uuid
WHERE   u.name = $permInOut.name
AND     (p.grant_on = $permInOut.grant_on OR p.grant_on = c.uuid)
AND     u.disabled = false
AND     u.removed = false

```